### PR TITLE
Add test cases for filters service

### LINKED ISSUE #295: Add tests for filters service
**Is your feature request related to a problem? Please describe.**
Filters service is a service introduced uniquely to watcher, it therefore does not have any test cases.

**Describe the solution you'd like**
Comprehensive test cases for the filters service.

### PR DESCRIPTION
**Pull request by Arif-Khalid**
### Summary:

Fixes part of #295 

#### Type of change:

- 🧪 Tests Update

### Changes Made:

- Added test cases for filters service
  - Covered initial behaviour
  - Covered clear filters
  - Covered update filters
  - Covered update preset view
- In order to reduce the size of the PR, other functionality will be tested for in another PR

### Proposed Commit Message:

```
Filters service does not have associated tests.

Tests for filters service would make the project
more stable.

Let's add test cases for the filters service.
```

<details><summary>
<h3>Checklist:</h3>
</summary>

- [x] I have tested my changes thoroughly.
- [x] I have created tests for any new code files created in this PR or provided a link to a issue/PR that addresses this.
- [x] I have added or modified code comments to improve code readability where necessary.
- [ ] I have updated the project's documentation as necessary.

</details>

### CODE REVIEW

**Comment by Arif-Khalid** [comment]
To be worked on after finalization of #310 as it makes significant changes to filters service.

File: src/app/core/services/filters.service.ts [hunk 1]
```diff
@@ -145,6 +145,7 @@ export class FiltersService {
   }
 
   clearFilters(): void {
+    this.updateFilters(this.defaultFilter);
```
Comment (nknguyenhc) [inline src/app/core/services/filters.service.ts:148]: What's the reasoning for adding this line? Isn't this already covered under the call to `updatePresetView`?

Comment (Arif-Khalid) [reply src/app/core/services/filters.service.ts:148]: After the most recent change to filters service, the preset view does not contain all filters anymore. Specifically in the current implementation, it excludes hiddenLabels. This ensures that the hidden labels filter and any future filter that is not bound to preset views are properly reset.

**Review by NereusWB922 — APPROVED** [review]
LGTM!

**Pull request merged**